src/range/accept_ranges.rs

```diff
@@ -0,0 +1,245 @@
+/// #
+/// use http_types::range::AcceptRanges;
+/// use http_types::Response;
+///
+/// let accept_ranges = AcceptRanges::new();
+///
+/// let mut res = Response::new(200);
+/// accept_ranges.apply(&mut res);
+///
+/// let accept_ranges = AcceptRanges::from_headers(res)?.unwrap();
+/// assert_eq!(accept_ranges.bytes(), false);
+/// assert_eq!(accept_ranges.other(), &None);
+/// #
+/// # Ok(()) }
+/// ```
+#[derive(Default, Debug, Clone, Eq, PartialEq)]
+pub struct AcceptRanges {
+    bytes: bool,
+    other: Option<String>,
+}
+
+impl AcceptRanges {
+    const BYTES: &'static str = "bytes";
+    const NONE: &'static str = "none";
+
+    /// Create a new AcceptRange which does not accept range requests.
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Create a new AcceptRange accepting `byte` unit.
+    pub fn with_bytes() -> Self {
+        AcceptRanges {
+            bytes: true,
+            ..Default::default()
+        }
+    }
+
+    /// Create a new AcceptRange accepting `byte` unit.
+    pub fn with_other(s: impl AsRef<str>) -> Self {
+        AcceptRanges {
+            other: Some(s.as_ref().to_owned()),
+            ..Default::default()
+        }
+    }
+
+    /// Returns true if AcceptRange accepts `byte` unit.
+    pub fn bytes(&self) -> bool {
+        self.bytes
+    }
+
+    /// Returns the supported unit if any.
+    pub fn other(&self) -> &Option<String> {
+        &self.other
+    }
+
+    /// Create a new instance from headers.
+    ///
+    /// Only a single AcceptRanges per resource is assumed to exist. If multiple Accept-Ranges
+    /// headers are found the last one is used.
+    pub fn from_headers(headers: impl AsRef<Headers>) -> crate::Result<Option<Self>> {
+        let headers = match headers.as_ref().get(ACCEPT_RANGES) {
+            Some(headers) => headers,
+            None => return Ok(None),
+        };
+
+        // If a header is returned we can assume at least one exists.
+        let s = headers.iter().last().unwrap().as_str();
+        Self::from_str(s).map(Some)
+    }
+
+    /// Create an AcceptRanges from a string.
+    pub(crate) fn from_str(s: &str) -> crate::Result<Self> {
+        let accept_range = match s {
+            Self::NONE => AcceptRanges::default(),
+            Self::BYTES => AcceptRanges::with_bytes(),
+            other => AcceptRanges::with_other(other),
+        };
+        Ok(accept_range)
+    }
+
+    /// Sets the `Accept-Ranges` header.
+    pub fn apply(&self, mut headers: impl AsMut<Headers>) {
+        headers.as_mut().insert(ACCEPT_RANGES, self.value());
+    }
+
+    /// Get the `HeaderName`.
+    pub fn name(&self) -> HeaderName {
+        ACCEPT_RANGES
+    }
+
+    /// Get the `HeaderValue`.
+    pub fn value(&self) -> HeaderValue {
+        let s = match self.other {
+            Some(ref other) => other.as_str(),
+            None if self.bytes => Self::BYTES,
+            None => Self::NONE,
+        };
```

Comment:
This could be factorized with the `Display` implementation. Let me known if you want that.